packages/react/src/components/ContentSwitcher/ContentSwitcher.tsx

```diff
@@ -238,6 +248,29 @@ export default class ContentSwitcher extends React.Component<
               selected: index === this.state.selectedIndex,
               ref: this.handleItemRef(index),
               size,
+              onMouseEnter: () => {
+                const { selectedIndex } = this.state;
+                // Check if the hovered index is adjacent to the selected button
+                if (Math.abs(index - selectedIndex!) === 1) {
+                  const selectedBtn = this._switchRefs[selectedIndex!];
+                  if (selectedBtn) {
+                    selectedBtn.classList.add(
+                      `${prefix}--content-switcher__selected-hovered`
+                    );
+                  }
+                }
+              },
+              onMouseLeave: () => {
+                const { selectedIndex } = this.state;
+                if (Math.abs(index - selectedIndex!) === 1) {
+                  const selectedBtn = this._switchRefs[selectedIndex!];
```

Comment:
Suggestion to make this implementation type safe.

```suggestion
                if (
                  typeof selectedIndex !== 'undefined' &&
                  Math.abs(index - selectedIndex) === 1
                ) {
                  const selectedBtn = this._switchRefs[selectedIndex];
```